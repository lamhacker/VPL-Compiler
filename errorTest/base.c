#include "base.h"

float* mallocVec(int num)
{
    float* vec = malloc(sizeof(float) * num + SSE_ALIGN);
	vec = (float *) align(vec, SSE_ALIGN);
	return vec;
}

void freeVec(float* vec)
{
    free(vec);
}

void setValue(float* vec, float val, int num)
{
    for (int i = 0; i < num; i++)
    {
        vec[i] = val;
    }
}

void printVec(char* rep, float* vec, int num)
{
	printf("%s: ", rep);
    for (int i = 0; i < num; i++)
    {   
        printf("%f ", vec[i]);
    }
	printf("\n");
}


