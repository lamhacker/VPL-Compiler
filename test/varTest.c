#include "base.h"
#define NUM (4)

extern void varOne(long, float*, float*);
extern void varTwo(long, float*, float*);
extern void varThree(long, float*, float*);



int main(void)
{
    float* a = mallocVec(NUM);
    float* b = mallocVec(NUM);
    
    printf("varOne: \n");
    setValue(a, 0, NUM);
    setValue(b, 0, NUM);
    varOne(NUM, a, b);
    printVec("a", a, NUM);
    printVec("b", b, NUM);
    
    printf("varTwo: \n");
    setValue(a, 0, NUM);
    setValue(b, 0, NUM);
    varTwo(NUM, a, b);
    printVec("a", a, NUM);
    printVec("b", b, NUM);
    
    printf("varThree: \n");
    setValue(a, 1, NUM);
    setValue(b, 3, NUM);
    varThree(NUM, a, b);
    printVec("a", a, NUM);
    printVec("b", b, NUM);
    
    
    freeVec(a);
    freeVec(b);
}
