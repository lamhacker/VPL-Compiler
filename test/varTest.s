
.text
.global varOne
.type varOne, @function
.p2align 4,,15

varOne:
    pushq   %rbp
    movq    %rsp, %rbp
    pushq   %rbx

    # allocate 0 local variables
    movq %rdi, %rax
    imulq $4, %rax, %rax
    addq $16, %rax
    imulq $0, %rax, %rax
    subq %rax, %rsp
    andq $-16, %rsp

    

    popq    %rbx
    leave
    ret

.text
.global varTwo
.type varTwo, @function
.p2align 4,,15

varTwo:
    pushq   %rbp
    movq    %rsp, %rbp
    pushq   %rbx

    # allocate 0 local variables
    movq %rdi, %rax
    imulq $4, %rax, %rax
    addq $16, %rax
    imulq $0, %rax, %rax
    subq %rax, %rsp
    andq $-16, %rsp

    

    popq    %rbx
    leave
    ret

.text
.global varThree
.type varThree, @function
.p2align 4,,15

varThree:
    pushq   %rbp
    movq    %rsp, %rbp
    pushq   %rbx

    # allocate 0 local variables
    movq %rdi, %rax
    imulq $4, %rax, %rax
    addq $16, %rax
    imulq $0, %rax, %rax
    subq %rax, %rsp
    andq $-16, %rsp

    

    popq    %rbx
    leave
    ret

