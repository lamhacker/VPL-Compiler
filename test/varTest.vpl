func varOne(a, b)
    var c;
end

func varTwo(a,b)
    var c, d;
end

func varThree(a,b)
    var c, d, e;
end
