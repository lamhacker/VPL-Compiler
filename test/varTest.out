varOne: 
a: 0.000000 0.000000 0.000000 0.000000 
b: 0.000000 0.000000 0.000000 0.000000 
varTwo: 
a: 0.000000 0.000000 0.000000 0.000000 
b: 0.000000 0.000000 0.000000 0.000000 
varThree: 
a: 1.000000 1.000000 1.000000 1.000000 
b: 3.000000 3.000000 3.000000 3.000000 
